pkg/metrics/prometheus_test.go

```diff
@@ -16,212 +16,90 @@
 package metrics
 
 import (
-	"fmt"
+	"context"
 	"net/http"
-	"net/http/httputil"
-	"reflect"
+	"net/http/httptest"
 	"strings"
 	"testing"
 	"time"
 
 	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/client_golang/prometheus/testutil"
 )
 
-func TestNewPrometheus(t *testing.T) {
-	test := struct {
-		name string
-		want Metrics
-	}{
-		name: "default",
-		want: &prom{
-			counters: make(map[string]prometheus.Counter),
-			summary:  make(map[string]prometheus.Summary),
-		},
+func TestMetrics(t *testing.T) {
+	ctx := WithMetrics(context.Background())
+	collector := FromContext(ctx)
+
+	_, err := collector.RegisterCounter(ctx, "test_counter", "label1")
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = collector.RegisterHistogram(ctx, "test_histogram", "label1")
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = collector.RegisterGauge(ctx, "test_gauge", "label1")
+	if err != nil {
+		t.Fatal(err)
 	}
-	t.Run(test.name, func(t *testing.T) {
-		if got := NewPrometheus(); !reflect.DeepEqual(got, test.want) {
-			t.Errorf("NewPrometheus() = %v, want %v", got, test.want)
-		}
-	})
 
-}
+	_, err = collector.RegisterGauge(ctx, "test_gauge", "label1")
+	if err == nil {
+		t.Fatal("expected error")
+	}
 
-func Test_prom_NewCounter(t *testing.T) {
-	type fields struct {
-		counters map[string]prometheus.Counter
-		summary  map[string]prometheus.Summary
+	time.Sleep(1 * time.Second)
```

Comment:
should have this probably be a bit more robust maybe a loop with the next lines to make sure we don't run into flakes (as much as possible)